packages/interactivity/src/directives.js

```diff
@@ -68,53 +72,48 @@ export default () => {
 		return createPortal( children, document.body );
 	} );
 
-	// data-wp-effect--[name]
-	directive( 'effect', ( { directives: { effect }, context, evaluate } ) => {
-		const contextValue = useContext( context );
-		Object.values( effect ).forEach( ( path ) => {
-			useSignalEffect( () => {
-				return evaluate( path, { context: contextValue } );
-			} );
+	// data-wp-watch--[name]
+	directive( 'watch', ( { directives: { watch }, evaluate } ) => {
+		watch.forEach( ( entry ) => {
+			useSignalEffect( () => evaluate( entry ) );
 		} );
 	} );
 
+	// data-wp-layout-init--[name]
+	directive(
+		'layout-init',
+		( { directives: { 'layout-init': layoutInit }, evaluate } ) => {
+			layoutInit.forEach( ( entry ) => {
+				useLayoutEffect( () => evaluate( entry ), [] );
+			} );
+		}
+	);
```

Comment:
As this comes from the copy/paste from Woo, let's remove it (we will remove it in Woo as well in the future).

Reply:
Done in https://github.com/WordPress/gutenberg/pull/55459/commits/c0664b6b1c7fb6589596c050392cbec5a8844fca.